.github/workflows/ci.yaml

```diff
@@ -8,6 +8,66 @@ on:
     - cron: '0 13 * * *'
 
 jobs:
+
+  chart-lint:
+    name: Helm chart Lint
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        kubernetes-version: [ "v1.17.17", "v1.22.0" ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: helm/chart-testing-action@v2.2.1
+
+      - name: Lint charts
+        run: ct --config .github/ct.yaml lint --debug
+
+      - name: Check for changed installable charts
+        id: list-changed
+        run: |
+          changed=$(ct --config .github/ct.yaml list-changed)
+          if [[ -n "$changed" ]]; then
+            echo "::set-output name=changed::true"
+          fi
+      - name: Run helm unit tests
+        if: steps.list-changed.outputs.changed == 'true'
+        run: |
+          helm plugin install https://github.com/quintush/helm-unittest
+          for chart in $(ct --config .github/ct.yaml list-changed); do
+            if [ -d "$chart/tests/" ]; then
+              helm unittest -3 $chart
+            else
+              echo "No unit tests found for $chart"
+            fi
+          done
+      - name: Setup Minikube
+        uses: manusa/actions-setup-minikube@v2.4.3
+        if: steps.list-changed.outputs.changed == 'true'
+        with:
+          minikube version: v1.20.0
+          kubernetes version: ${{ matrix.kubernetes-version }}
+          github token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/setup-go@v2
+        if: steps.list-changed.outputs.changed == 'true'
+        with:
+          go-version: '1.18'
+      - name: Create image for chart testing
+        if: steps.list-changed.outputs.changed == 'true'
+        run: |
+          GOOS=linux GOARCH=amd64 make build # Set GOOS and GOARCH explicitly since Dockerfile expects them in the binary name
+          DOCKER_BUILDKIT=1 docker build -t e2e/newrelic-infra-operator:test  .
+          minikube image load e2e/newrelic-infra-operator:test
+          kubectl create ns ct
+      - name: Test install charts
+        if: steps.list-changed.outputs.changed == 'true'
+        run: ct install --namespace ct --config .github/ct.yaml --debug
+      - name: Test upgrade charts
+        if: steps.list-changed.outputs.changed == 'true'
+        run: ct install --namespace ct --config .github/ct.yaml --debug --upgrade
```

Comment:
I know that I am saying this like 3 PRs before we were merging this pattern but I thought about this right now...

WDYT if we have the upgradability test separated in a separate job? In case we have to change the configuration it could fail and we can ignore it. Or even add a tag like `breaking-chart` and don't run the chart at all.

I would not like you to tackle this right now, just thinking out loud.

Reply:
Yes whenever we change the config the upgrade process could fail. I agree that is not ideal, but testing the upgrade process is important. 

Since it will rarely happen to have a configuration breaking change we can test it manually and ignore the job output